ads/google/a4a/traffic-experiments.js

```diff
@@ -81,35 +92,59 @@ export function googleAdsIsA4AEnabled(win, element, experimentId, branches) {
+ * Set experiment state from URL parameter, if present.  This looks for the
+ * presence of a URL parameter of the form
+ *   `exp=expt0:val0,expt1:val1,...,a4a:X,...,exptN:valN`
+ * and interprets the X as one of the following:
+ *   - `-1`: Manually-triggered experiment.  For testing only.  Sets
+ *     `adtest=on` on the ad request, so that it will not bill or record
+ *     user clicks as ad CTR.  Ad request will be accounted in a special
+ *     'testing only' experiment statistic pool so that we can track usage
+ *     of this feature.
+ *   - `0`: Ad is explicitly opted out of the overall A4A-vs-3p iframe
+ *     experiment.  Ad will serve into a 3p iframe, as traditional, but ad
+ *     request and clicks will not be accounted in experiment statistics.
+ *   - `1`: Ad is on the control branch of the overall A4A-vs-3p iframe
+ *     experiment.  Ad will serve into a 3p iframe, and ad requests and
+ *     clicks _will_ be accounted in experiment statistics.
+ *   - `2`: Ad is on the experimental branch of the overall A4A-vs-3p iframe
+ *     experiment.  Ad will render via the A4A path, including early ad
+ *     request and (possibly) early rendering in shadow DOM or iframe.
  *
- * @param {!Window} win
- * @param {string} experimentId
- * @param {!Branches} branches
- * @visibleForTesting
+ * @param {!Window} win  Window.
+ * @param {!string} experimentName  Name of the overall experiment.
+ * @param {!string} controlBranchId  Experiment ID string for control branch of
+ *   the overall experiment.
+ * @param {!string} treatmentBranchId  Experiment ID string for the 'treatment'
+ *   (i.e., a4a) branch of the overall experiment.
+ * @param {!string} manualId  ID of the manual experiment.
  */
-function handleUrlParameters(win, experimentId, branches) {
-  const a4aParam = /(?:\?|&)a4a=([0-9]+)/.exec(win.location.search);
-  if (a4aParam) {
-    // If a4aParam is non-null, it necessarily has at least 2 elements.
-    switch (a4aParam[1]) {
-      case '0':
-        // Not selected into experiment.  Disable the experiment altogether, so
-        // that setupPageExperiments doesn't accidentally enable it.
-        forceExperimentBranch(win, experimentId, false);
-        break;
-      case '1':
-        // Selected in; on control branch.
-        forceExperimentBranch(win, experimentId, branches.control);
-        break;
-      case '2':
-        // Selected in; on experiment branch.
-        forceExperimentBranch(win, experimentId, branches.experiment);
-        break;
-      default:
-        dev().warn('a4a-config', 'Unknown a4a URL parameter: ',
-                 a4aParam[1]);
-    }
+function handleUrlParameters(win, experimentName,
+    controlBranchId, treatmentBranchId, manualId) {
+  const expParam = parseQueryString(win.location.search)['exp'];
+  if (!expParam) {
+    return;
+  }
+  const a4aParam = expParam.split(',').find(
+      x => { return x.substr(0, 4) == 'a4a:'; });
```

Comment:
`return x.indexOf('a4a:') == 0` might be better, so that we avoided the magic number `4`.